Fix local Snuba Admin import errors by adding codemirror packages

onkar opened the pull request:
`make watch-admin `on local shows up a bunch of errors like this:
```
ERROR in /Users/onkardeshpande/code/snuba/snuba/admin/static/common/components/sql_editor.tsx
./static/common/components/sql_editor.tsx 6:20-42
[tsl] ERROR in /Users/onkardeshpande/code/snuba/snuba/admin/static/common/components/sql_editor.tsx(6,21)
      TS2307: Cannot find module '@codemirror/lang-sql' or its corresponding type declarations.
 @ ./static/query_editor.tsx 40:19-69
 @ ./static/snuba_explain/index.tsx 51:37-71
 @ ./static/data.tsx 20:38-73
 @ ./static/index.tsx 35:13-39
``` 
This change fixes it by adding these modules in dev dependencies.

gggritso commented:
🤔 hmm, this doesn't seem right! `devDependencies` is only for things that are _only_ needed in development (Webpack, Jest, etc.). CodeMirror is a production dependency, it shouldn't be in `devDependencies`. `yarn install` installs both production and development dependencies.

Are you sure you didn't just need to run `yarn install` again? Watch out, you have to run it in the `snuba/admin/` directory!

onkar commented:
> 🤔 hmm, this doesn't seem right! `devDependencies` is only for things that are _only_ needed in development (Webpack, Jest, etc.). CodeMirror is a production dependency, it shouldn't be in `devDependencies`. `yarn install` installs both production and development dependencies.
> 
> Are you sure you didn't just need to run `yarn install` again? Watch out, you have to run it in the `snuba/admin/` directory!

After sending this PR for review, I had `@codemirror` and `@lezer` in my `snuba/admin/node_modules` directory. I did the following things:

1. Remove the code I added in `devDependencies` section.
2. Rename `snuba/admin/node_modules/@codemirror` to `snuba/admin/node_modules/@codemirror-orig` and `snuba/admin/node_modules/@lezer` to `snuba/admin/node_modules/@lezer-orig`
3. I started seeing red lines in vscode around the `import` statements.
4. Ran `yarn install` from `snuba/admin`:
```
(.venv) onkardeshpande@G260FHW6HW admin % yarn install
yarn install v1.22.22
warning package.json: No license field
warning snuba-admin@1.0.0: No license field
[1/4] 🔍  Resolving packages...
success Already up-to-date
```
which did not pull in `@codemirror` or `@lezer` packages.
5. Went in the topmost snuba directory i.e. `/Users/onkardeshpande/code/snuba` and did this:
```
(.venv) onkardeshpande@G260FHW6HW snuba % make watch-admin
cd snuba/admin && yarn install && yarn run watch
```
which does run `yarn install && yarn run watch` by cding to `snuba/admin` but got the errors in the PR description instead of pulling required packages.

So it looks like manually running `yarn install` or running `make watch-admin` does not generate these packages, unless I am doing something wrong. So, that is why I added these packages in the `devDependencies` section.

Surprisingly, I did not add `@lezer` in my PR and yet that package was pulled in successfully.

Any thoughts?

gggritso commented:
@onkar something is definitely not right!

First of all, what's the output of these commands?

- `which node` (should be `"/Users/onkardeshpande/.volta/bin/node"`)
- `node -v` (should be `"v20.8.1"`)
- `which yarn` (should be `"/Users/onkardeshpande/.volta/bin/yarn"`)
- `yarn -v` (should be `"1.22.19"`)

It looks like your yarn is version is `"v1.22.22"` which is not right! Do you have [Volta](https://volta.sh) installed? If not, you'll need to do that! 

Second of all, sometimes `node_modules` packages fall out of sync in surprising ways. I recommend at this point you run `rm -rf node_modules` in `snuba/admin` and clear the directory out, and run `yarn install` again (after making sure you have working correct versions of Volta, node, and yarn)

And lastly, it makes sense that `lezer` is available, since it's a dependency of `@codemirror` 👍🏻

onkar commented:
This PR is already closed but just for the record, `rm -rf snuba/admin/node_modules` followed by `make watch-admin` from `snuba` directory fixed it for me. Although, using `volta` was probably a better solution.
